#FBSD7+amd64+gcc421 is okay without the patch (w/o -lsupc++ is fine)
#FBSD6+amd64+gcc342 is okay with the patch    (w/o -lsupc++ is fine)
 ...-lsupc++ is static, and cannot find -lgcc_s
#FBSD6+i386+gcc342  is okay with the patch    (w/o -lsupc++ -lgcc_s is fine)
    cannot find -lgcc_s

--- store/util/makefile.mk.orig	2007-12-12 22:23:13.000000000 +0900
+++ store/util/makefile.mk	2008-08-16 20:52:44.000000000 +0900
@@ -76,6 +76,7 @@
 .ENDIF
 
 # On gcc3 __Unwind_SetIP is not in supc++ but in libgcc_s.so
+# FBSD6+amd64+gcc342 does not require -lsupc++. only libsupc++.a is supplied
 .IF "$(COMID)"=="gcc3" && "$(GUI)"!="OS2"
 .IF "$(GUI)"=="WNT"
 SHL1STDLIBS+= 	-lsupc++
@@ -86,6 +87,8 @@
 .IF "$(CCNUMVER)"<="000399999999"
 SHL1STDLIBS+=	-lsupc++
 .ENDIF # CCNUMVER
+.ELIF "$(OS)"=="FREEBSD"
+SHL1STDLIBS+=
 .ELIF "$(CCNUMVER)"<="000400000999"
 SHL1STDLIBS+= 	-lsupc++ -lgcc_s
 .ENDIF
